src/car/car.service.ts

```diff
@@ -93,21 +98,50 @@ export class CarService {
                         count: 1,
                     },
                 };
+                acc[`${monthName[month]},${year}`][car.car_id].percentage = ((acc[`${monthName[month]},${year}`][car.car_id].days / 30) * 100).toFixed(2);
             }
 
             return acc;
         }, {});
 
-        const sortedAllCarsUsage = Object.keys(allCarsUsage)
+        const sortedAllCarsUsage = Object.entries(allCarsUsage)
             .sort((a, b) => {
-                return monthName.indexOf(a) - monthName.indexOf(b);
+                const [monthA, yearA] = a[0].split(',');
+                const [monthB, yearB] = b[0].split(',');
+                if (yearA < yearB) {
+                    return -1;
+                } else if (yearA > yearB) {
+                    return 1;
+                } else {
+                    if (monthName.indexOf(monthA) < monthName.indexOf(monthB)) {
+                        return -1;
+                    } else if (monthName.indexOf(monthA) > monthName.indexOf(monthB)) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                }
             })
-            .reduce((accumulator, key) => {
-                accumulator[key] = allCarsUsage[key];
-
-                return accumulator;
+            .reduce((acc, [key, value]) => {
+                acc[key] = value;
+                return acc;
             }, {});
 
+        const toFile = Object.entries(sortedAllCarsUsage).map(([key, value]) => {
+            return {
+                month: key,
+                cars: Object.entries(value).map(([key, value]) => {
+                    return {
+                        car_id: key,
+                        ...value,
+                    };
+                }),
+            };
+        });
+
+        const csv = new ObjectsToCsv(toFile);
+        await csv.toDisk('./carsUsage.csv');
```

Comment:
I want to have both options csv and json

Reply:
Feature added.